### PR TITLE
Add new flags and memcached support to query frontend

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

This pr supersedes #163.

Added new flag configurations for the labels and series middleware of the query frontend.

### CODE REVIEW

File: all.jsonnet
Comment (metalmatze) [inline all.jsonnet:115]: These were changed on purposed to different values than the defaults, to show that overwriting the values actually works, which is kind of the entire point of the `all.jsonnet` in total :wink: :blush: 
Happy to add this as a comment to the top of the file. Wdyt?

**Review by metalmatze — COMMENTED** [review]
One comment, otherwise looks really good :+1:

**Review by kakkoyun — COMMENTED** [review]
LGTM 🥇 

Could you update Thanos version that supports these changes? These aren't in v0.16.0, are they?

File: all.jsonnet
Comment (yeya24) [reply all.jsonnet:115]: Oh, got it. Thanks for the clarification! 
I was wondering why this changed yesterday :rofl: I can update these values back.

**Comment by yeya24** [comment]
> LGTM 
> 
> Could you update Thanos version that supports these changes? These aren't in v0.16.0, are they?

They are part of the new release so v0.17.0. Can I use the current master image?

File: all.jsonnet
Comment (metalmatze) [reply all.jsonnet:115]: Yes, that would be great and maybe even change values in `all.jsonnet` to different ones that the defaults, haha. :) Doesn't need to be too crazy though.

**Comment by metalmatze** [comment]
Then let's wait for a couple of hours / days with merging this and update to at least use the first rc I'll cut soon.

**Review by bwplotka — APPROVED** [review]
Nice,

**Comment by bwplotka** [comment]
>Then let's wait for a couple of hours / days with merging this and update to at least use the first rc I'll cut soon.

This does not mean we cannot use it in our prod/stage conf code, right? (:

**Comment by metalmatze** [comment]
Hm, good point about using these new features downstream. I guess for that reason we could already go ahead and merge it, since I usually see this master branch match mostly whatever Thanos master is doing. 
The overhead of adding this to our downstream for 2 days is probably not justified. 
Feel free to merge in this case!

**Comment by metalmatze** [comment]
Sorry, I quickly want to release v0.16.0 branch without this. Happy to merge after that (next 30min). Just making sure we don't merge it before that, thus converting to draft :innocent:

**Comment by yeya24** [comment]
I guess it is ready to go. Mark this ready for review.

**Pull request merged**
